### PR TITLE
[optimizations] Use visitor pattern in remove-unused

### PR DESCRIPTION
**Pull request by danmaninc**

### CODE REVIEW

File: src/analyzer/strategies/remove-unused.hpp [hunk 1]
```diff
@@ -1,19 +1,232 @@
...
+
+                this->table.emplace(var.m_name, 0);
+                continue;
+            }
+
+            // here we calculate the usages of vars
+            stmt->accept(*this);
+        }
+
+        std::unordered_map<std::string, int> variables = this->table;
```
Comment (danmaninc) [inline src/analyzer/strategies/remove-unused.hpp]: i'm confused by this line
i think that it could be much better than it is right now.

Comment (Processor228) [reply src/analyzer/strategies/remove-unused.hpp]: yeah i see

Comment (Processor228) [reply src/analyzer/strategies/remove-unused.hpp]: i've fixed some stuff there, now code looks more logical

Comment (Processor228) [reply src/analyzer/strategies/remove-unused.hpp]: well now this MR has my approve✅

**Pull request merged**